### ISSUE TITLE
Cannot read property 'Symbol(Symbol.iterator)'

### ISSUE THREAD

**Issue by vhpoet** [interledgerjs/ilp-kit#129]
```
[api] (node:28062) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
```

**Comment by Winthan** [comment]
it is happening here too.

**Comment by sharafian** [comment]
By registering the event `process.on('unandledRejection')`, I was able to get a stacktrace for this error. Unfortunately it's a pretty opaque stack trace. I went into the `koa.io` code to see what was going on, but I didn't get anywhere useful. It's hard to follow the structure of the code with all the generators and anonymous functions being passed around.


```
TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
[api]     at Socket.<anonymous> (/home/sharafian/ilp-kit/node_modules/koa.io/lib/socket.io/router.js:104:15)
[api]     at next (native)
[api]     at Socket.<anonymous> (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/koa-compose/index.js:29:12)
[api]     at next (native)
[api]     at onFulfilled (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:61:19)
[api]     at /home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:50:5
[api]     at Socket.co (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:49:10)
[api]     at Socket.createPromise (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:30:15)
[api]     at Socket.socket.onevent (/home/sharafian/ilp-kit/node_modules/koa.io/lib/socket.io/router.js:63:20)
[api]     at Socket.onpacket (/home/sharafian/ilp-kit/node_modules/socket.io/lib/socket.js:290:12)
[api]     at Client.ondecoded (/home/sharafian/ilp-kit/node_modules/socket.io/lib/client.js:193:14)
[api]     at Decoder.Emitter.emit (/home/sharafian/ilp-kit/node_modules/component-emitter/index.js:134:20)
[api]     at Decoder.add (/home/sharafian/ilp-kit/node_modules/socket.io-parser/index.js:247:12)
[api]     at Client.ondata (/home/sharafian/ilp-kit/node_modules/socket.io/lib/client.js:175:18)
[api]     at emitOne (events.js:96:13)                                                                                                                                                                                                 [api]     at Socket.emit (events.js:188:7) Promise {
[api]   <rejected> TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined
[api]     at Socket.<anonymous> (/home/sharafian/ilp-kit/node_modules/koa.io/lib/socket.io/router.js:104:15)                                                                                                                           [api]     at next (native)
[api]     at Socket.<anonymous> (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/koa-compose/index.js:29:12)
[api]     at next (native)                                                                                                                                                                                                             [api]     at onFulfilled (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:61:19)
[api]     at /home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:50:5
[api]     at Socket.co (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:49:10)
[api]     at Socket.createPromise (/home/sharafian/ilp-kit/node_modules/koa.io/node_modules/co/index.js:30:15)
[api]     at Socket.socket.onevent (/home/sharafian/ilp-kit/node_modules/koa.io/lib/socket.io/router.js:63:20)
[api]     at Socket.onpacket (/home/sharafian/ilp-kit/node_modules/socket.io/lib/socket.js:290:12)
[api]     at Client.ondecoded (/home/sharafian/ilp-kit/node_modules/socket.io/lib/client.js:193:14)
[api]     at Decoder.Emitter.emit (/home/sharafian/ilp-kit/node_modules/component-emitter/index.js:134:20)
[api]     at Decoder.add (/home/sharafian/ilp-kit/node_modules/socket.io-parser/index.js:247:12)
[api]     at Client.ondata (/home/sharafian/ilp-kit/node_modules/socket.io/lib/client.js:175:18)
[api]     at emitOne (events.js:96:13)
[api]     at Socket.emit (events.js:188:7) }
```

**Comment by Winthan** [comment]
I am getting this.
`[api] (node:8768) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined`
